### ISSUE TITLE
Node dumps core when the stored procedure returns an uninitialized reference cursor

### ISSUE THREAD

**Issue by Buto** [oracle/node-oracledb#170]
node-oracledb version 0.7
Scenario:
1. A Oracle Stored procedure returns a reference cursor in its parameters.
2. The stored procedure raises an exception before the  "OPEN p_parameter FOR select * from foo;" executes.
3. Node dumps core after this sp call returns.
end of scenario.  

The following code fragment should reproduce the defect:
```sql
CREATE OR REPLACE
 PROCEDURE foo( p_details   IN OUT  SYS_REFCURSOR )
   IS
   BEGIN
      --
     RAISE DUP_VAL_ON_INDEX;  -- exception raised before reference cursor is initialized
      -- 
      OPEN p_details FOR
         SELECT   * FROM     employees;
      --
   EXCEPTION
        WHEN OTHERS
             NULL;
   END;
/
```

**Comment by cjbj** [comment]
Thanks @Buto

**Comment by cjbj** [comment]
@Buto this is fixed in 1.1
